app/controllers/application_controller.rb

```diff
@@ -5,6 +5,137 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "garland"
+  end
+
+  get '/' do 
+    erb :index
+  end
+
+  get '/users/:slug' do 
+    @user = User.find_by_slug(params[:slug])
+    @tweets = Tweet.all.find { |tweet| tweet.user_id == @user.id }
+    erb :'tweets/tweets'
+  end
+
+  get '/tweets' do
```

Comment:
This controller is overloaded. Consider separating your concerns by splitting some of these actions into separate controllers (i.e. a users controller and a tweets controller).